packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo } from 'react';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
+
+import { mergeClasses } from '../../classify';
+import CategorySortItem from './categorySortItem';
+import defaultClasses from './categorySort.css';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+    const { elementRef, expanded, setExpanded } = useDropdown();
+
+    const sortElements = useMemo(() => {
+        const onMenuItemClick = sortAttribute => {
+            setSort({
+                sortAttribute: sortAttribute.attribute,
+                sortDirection: sortAttribute.sortDirection
+            });
+            setExpanded(false);
+        };
+
+        // should be not render item in collapsed mode.
+        if (expanded === false) {
+            return;
```

Comment:
One minor thing here: we tend to return `null` in these cases (rather than `undefined`).
```suggestion
            return null;
```